exporter/trace/trace_proto.go

```diff
@@ -71,11 +71,29 @@ func generateDisplayName(s *export.SpanData, format DisplayNameFormatter) string
 	}
 }
 
+func injectLabelsFromResources(sd *export.SpanData) {
+	if sd.Resource.Len() == 0 {
+		return
+	}
+	uniqueAttrs := make(map[kv.KeyValue]bool)
```

Comment:
AFAIU the goal is to have a set of unique keys, and skip resource attributes which conflict with the span attributes.
If so, you should use a string key as a key in this map, rather than entire KeyValue struct.
Please correct me if I misunderstood the purpose of this map.

Reply:
You can also preallocate the map capacity as
```
make(map[...]bool, len(sd.Attributes)
```
since we now that we will at least add all span attributes to the map

Reply:
Sounds good. Will do